crates/bevy_math/src/sampling/statistical_tests/impls.rs

```diff
@@ -0,0 +1,408 @@
+            partition_range(0., PI, self.polar_bins)
+                .into_iter()
+                .map(|psi| 1. - psi.cos())
+                .collect::<Vec<_>>(),
+        );
+
+        [bins_radial, bins_azimuthal, bins_polar]
+    }
+
+    fn dfs(&self) -> [usize; 3] {
+        [
+            self.radial_bins.saturating_sub(1),
+            self.azimuthal_bins.saturating_sub(1),
+            self.polar_bins.saturating_sub(1),
+        ]
+    }
+}
+
+/// A discretized distribution for the boundary of a [`Sphere`].
+#[derive(Clone, Copy)]
+pub struct SphereBoundaryBins {
+    sphere: BoundaryOf<Sphere>,
+    azimuthal_bins: usize,
+    polar_bins: usize,
+}
+
+impl SphereBoundaryBins {
+    /// Create a new discretized distribution for the boundary of the given `sphere`, with `azimuthal_bins`
+    /// swept out along the azimuth and `polar_bins` swept out along the polar angle.
+    pub fn new(sphere: Sphere, azimuthal_bins: usize, polar_bins: usize) -> Self {
+        Self {
+            sphere: sphere.boundary_dist(),
+            azimuthal_bins,
+            polar_bins,
+        }
+    }
+}
+
+impl Binned<2> for SphereBoundaryBins {
+    type IntermediateValue = Vec3;
+    type InnerDistribution = BoundaryOf<Sphere>;
+    fn inner_dist(&self) -> Self::InnerDistribution {
+        self.sphere
+    }
+
+    fn bin(&self, value: Vec3) -> Option<[usize; 2]> {
+        let theta = value.xy().to_angle();
+        let psi = value.z.acos();
+
+        if !theta.is_finite() || !psi.is_finite() {
+            None
+        } else {
+            let azimuthal_bin = bin_from_range(theta, -PI, PI, self.azimuthal_bins)?;
+            let polar_bin = bin_from_range(psi, 0., PI, self.polar_bins)?;
+            Some([azimuthal_bin, polar_bin])
+        }
+    }
+}
+
+impl WithBinDistributions<2> for SphereBoundaryBins {
+    fn get_bins(&self) -> [BinDistribution; 2] {
+        let bins_azimuthal = BinDistribution::from_weights(vec![
+            1. / self.azimuthal_bins as f32;
+            self.azimuthal_bins
+        ]);
+
+        // Factor out 2 * pi from the surface area here for simplicity
+        let bins_polar = BinDistribution::from_cdf(
+            partition_range(0., PI, self.polar_bins)
+                .into_iter()
+                .map(|psi| 1. - psi.cos())
+                .collect::<Vec<_>>(),
+        );
+
+        [bins_azimuthal, bins_polar]
+    }
+
+    fn dfs(&self) -> [usize; 2] {
+        [
+            self.azimuthal_bins.saturating_sub(1),
+            self.polar_bins.saturating_sub(1),
+        ]
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::{super::stats::*, *};
+    use rand::{distributions::Distribution, rngs::StdRng, SeedableRng};
+
+    /// Run goodness-of-fit tests for all directional components of this distribution.
+    fn test_components<const N: usize, T>(binned: T, samples: usize)
+    where
+        T: Binned<N> + WithBinDistributions<N> + Copy,
+    {
+        let rng = StdRng::from_entropy();
```

Comment:
I don't think we should be using unseeded RNG in our automated tests unfortunately.